nodebuilder/header/service.go

```diff
@@ -51,6 +52,33 @@ func (s *Service) GetVerifiedRangeByHeight(
 }
 
 func (s *Service) GetByHeight(ctx context.Context, height uint64) (*header.ExtendedHeader, error) {
+	// TODO(vgonkivs): remove after https://github.com/celestiaorg/go-header/issues/32 will be
+	// implemented
+	head, err := s.store.Head(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if uint64(head.Height()) < height {
+		return nil, fmt.Errorf("header: given height exceeds local head."+
```

Comment:
```suggestion
		return nil, fmt.Errorf("header: node is syncing"+
```

Reply:
Let's make it clear for the user what the error means. As before it seems like the input was wrong